iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -110,6 +110,40 @@ Viewer.prototype._addDocumentSetTabs = function() {
     }
 }
 
+/*
+ * Rewrite hyperlinks in the iXBRL.
+ *
+ * Relative links to other files in the same document set are handled by
+ * JavaScript to switch tabs within the viewer
+ *
+ * All other links are forced to open in a new tab
+ *
+ */
+Viewer.prototype._updateLink = function(n) {
+    const url = $(n).attr("href");
+    if (url !== undefined) {
+        const [file, fragment] = url.split('#', 2);
+        const docSetFiles = this._report.documentSetFiles();
+        if (this._report.isDocumentSet() && !url.includes('/') && docSetFiles.includes(file)) {
+            $(n).click((e) => { 
+                const index = docSetFiles.indexOf(file);
+                this.selectDocument(index); 
+                if (fragment !== undefined && fragment != "") {
+                    const f = $.escapeSelector(fragment);
```

Comment:
I wonder if this should retry after precent-decoding the fragment (per [HTML5 spec](https://html.spec.whatwg.org/multipage/browsing-the-web.html#scroll-to-the-fragment-identifier), specifically steps 4+5 of [this section](https://html.spec.whatwg.org/multipage/browsing-the-web.html#the-indicated-part-of-the-document)).  Well, the code here seems like a good improvement anyway, and we can always revisit if it causes a problem for an actual document.

Reply:
If we're doing it, we may as well do it right - done in 5d61b50.  I managed to convince myself that `decodeURIComponent` does the right thing with UTF-8 percent encoded strings.